### PR TITLE
Initial ORD Service implementation

### PR DESCRIPTION
**Pull request by NickyMateev**
# Initial ORD Service implementation

The initial code base of the ORD Service.

### CODE REVIEW

File: components/ord-service/Makefile [hunk 1]
```diff
@@ -0,0 +1,32 @@
+APP_NAME = compass-ord-service
+APP_PATH = components/ord-service
+BUILDPACK = eu.gcr.io/kyma-project/test-infra/pr/buildpack-java:PR-3076
```
Comment (NickyMateev) [inline components/ord-service/Makefile:3]: Set proper image tag after this is merged: https://github.com/kyma-project/test-infra/pull/3076

File: components/ord-service/src/test/java/com/sap/cloud/cmp/ord/service/filter/aggregator/JsonArrayElementsAggregatorTest.java [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+
+        JsonNode jsonTree = mapper.readTree(content);
+        aggregator.aggregate(jsonTree);
+        String actualContent = mapper.writeValueAsString(jsonTree);
+
+        assertEquals(expectedContent, actualContent);
+    }
+
+    @Test
+    public void testAggregate_ReturnsUnmodifiedJson_WhenLabelsFound() throws Exception {
```
Comment (DimitarPetrov) [inline components/ord-service/src/test/java/com/sap/cloud/cmp/ord/service/filter/aggregator/JsonArrayElementsAggregatorTest.java:56]: Wrong test name. 

Also, I think it makes sense to have tests for nested resources for example labels and arrays in API which is part of a package.

**Review by DimitarPetrov — CHANGES_REQUESTED** [review]

File: components/ord-service/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,43 @@
+FROM maven:3.6.3-jdk-8-slim AS builder
+
+ENV BASE_APP_DIR /go/src/github.com/kyma-incubator/compass/components/ord-service
+ENV COMPONENT_DIR /go/src/github.com/kyma-incubator/compass/components/ord-service
```
[the remark this answers is not in the document]
Comment (NickyMateev) [reply components/ord-service/Dockerfile:4]: The second one is used in the `RUN ./scripts/install_dependencies.sh && ./scripts/build.sh -DskipTests && \` step

File: components/ord-service/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+
+#
+# Build app
+#
+
+RUN ./scripts/install_dependencies.sh && ./scripts/build.sh -DskipTests && \
+    VERSION=$(mvn org.apache.maven.plugins:maven-help-plugin:3.2.0:evaluate -Dexpression=project.version -DforceStdout -q) && \
+    mkdir /app && mv ./target/ord-service-$VERSION.jar /app/ord-service.jar
+
+FROM openjdk:8
```
[the remark this answers is not in the document]
Comment (NickyMateev) [reply components/ord-service/Dockerfile:23]: I don't think there's a widely distributed openjre image

File: components/ord-service/src/test/java/com/sap/cloud/cmp/ord/service/filter/aggregator/JsonArrayElementsAggregatorTest.java (hunk 2 above)
Comment (NickyMateev) [reply components/ord-service/src/test/java/com/sap/cloud/cmp/ord/service/filter/aggregator/JsonArrayElementsAggregatorTest.java:56]: Yep, that's a good idea!

File: components/ord-service/scripts/install_dependencies.sh [hunk 5]
```diff
@@ -0,0 +1,23 @@
+#!/usr/bin/env bash
+
+OLINGO_JPA_LIB_DIR="$COMPONENT_DIR/olingo-jpa-processor-v4"
+OLINGO_VERSION_TAG="0.3.8"
```
[the remark this answers is not in the document]
Comment (NickyMateev) [reply components/ord-service/scripts/install_dependencies.sh]: Okay, so in order to have the pom.xml as source of truth I made this script extract the version with `sed` from the pom.xml. If you have a better idea, please say so. 😄

**Review by DimitarPetrov — APPROVED** [review]

**Pull request merged**